test/controllers/CancelDeclarationControllerSpec.scala

```diff
@@ -64,21 +70,28 @@ class CancelDeclarationControllerSpec extends ControllerWithoutFormSpec with Err
     "return 200 (OK)" when {
 
       "display page method is invoked" in new SetUp {
-        val result = controller.displayPage(getRequestWithSession(sessionData.toSeq: _*))
+        val result = controller.displayPage(additionalDeclarationType)(getRequestWithSession(sessionData.toSeq: _*))
         status(result) must be(OK)
       }
 
       "cancellation is requested with duplicate request error" in new SetUp {
-        cancelDeclarationResponse(CancellationAlreadyRequested)
-        val result = controller.onSubmit()(postRequestWithSession(correctCancelDeclarationJSON, sessionData.toSeq))
+        cancelDeclarationResponse(CancellationResult(CancellationAlreadyRequested, Some("conversationId")))
+
+        when(mockAuditService.auditAllPagesDeclarationCancellation(any())(any())).thenReturn(Future.successful(Success))
```

Comment:
Maybe add a `verify` statement to prove that the audit service's method is called as expected